make sure SDK docs pages have titles

jdamiba opened the pull request:
The purpose of this PR is to address the issue raised by https://github.com/daytonaio/docs/issues/298. 

Currently, when users visit pages such as https://www.daytona.io/docs/python-sdk/daytona/ or https://www.daytona.io/docs/typescript-sdk/daytona/, their browser tab bar shows `| Daytona`. 

This is because when the documentation markdown files are created, the title field of the front-matter is explicitly set to be an empty string: https://github.com/daytonaio/sdk/blob/d716ae4ff0a476059d4d81e123dc4a2a4b3f876c/packages/typescript/hooks/typedoc-custom.mjs#L11 and  https://github.com/daytonaio/sdk/blob/d716ae4ff0a476059d4d81e123dc4a2a4b3f876c/packages/python/pydoc-markdown.yml#L13

When these markdown files are converted into the web app that serves the documentation, this empty string is being passed as the value of the title field for that page, along with a `| Daytona` decorator that is appended to all page titles. 

In order to ensure that each generated Markdown page has a relevant title, I have made two changes. On the Python side, I changed the `sed` command to use the very convenient `FIRST_SECTION` environment variable that is passed to every invocation of `pydoc-markdown` command. 

On the Typescript side it was a bit more complicated because the `typedoc` command that generates the Markdown documentation files is not passed a handy environment variable with a relevant name. However, once the Markdown file is generated, we can just grab the value of the first heading and use that as the title. 

I tested this PR by executing `npm run docs` in my local environment and ensuring that each generated Markdown file had a relevant title in the front matter.

MDzaja reviewed (changes requested):
@jdamiba There are two improvements I would suggest:
1. Use the filename as the title, but convert it to CamelCase format. For example, if the filename is `execute-response.mdx`, the title should be `ExecuteResponse`. In most cases, this will produce the same title as your current solution, but it will correct edge cases like `errors.mdx`.
2. Add `hideTitleOnPage: true` to the frontmatter of each file. We'll update the docs repo to hide the title from the page content when this flag is set, and display it only in the tab bar.

Example of the current frontmatter and the resulting page:
```
---
title: "DaytonaError"
---
```

<img width="1325" alt="image" src="https://github.com/user-attachments/assets/461446d5-1871-4188-b4f1-e95b7d4e55f2" />

Example of the desired frontmatter and the resulting page:
```
---
title: "Errors"
hideTitleOnPage: true
---
```
<img width="1311" alt="image" src="https://github.com/user-attachments/assets/8a7a1497-6315-48f8-8c87-f89cb164b425" />

jdamiba commented:
I have updated my PR with the suggested improvements. 

Firstly, I have added `hideOnPage: true` to the front-matter each generated Markdown file. 

Second, we now generate titles for each markdown page based on the file name. We capitalize the first letter of each word, with the hyphen as the delimiter.

MDzaja reviewed (approved):
Looks good to me.
